locksmith/src/models/jobs.ts

```diff
@@ -0,0 +1,51 @@
+import KeyData from './keyData'
+
+type JobStatus = 'processing' | 'completed' | 'failed'
+
+interface Job {
+  id: string
+  status: JobStatus
+  createdAt: number
+  // Can hold result data or error message
+  data?: KeyData[] | string
+}
+
+class JobStore {
```

Comment:
We need a better name here because we will not remember what kind of jobs we are talking about here.

What about `DownloadAllKeysJobStore` for example?